packages/neos-ui/src/Containers/Modals/SelectNodeType/index.js

```diff
@@ -15,7 +15,13 @@ import NodeTypeGroupPanel from './nodeTypeGroupPanel';
 import NodeTypeFilter from './nodeTypeFilter';
 import style from './style.css';
 
-const calculateInitialMode = (allowedSiblingNodeTypes, allowedChildNodeTypes) => {
+const calculateInitialMode = (allowedSiblingNodeTypes, allowedChildNodeTypes, preferredMode) => {
+    if (
+        ((preferredMode === 'before' || preferredMode === 'after') && allowedSiblingNodeTypes.length) ||
+        (preferredMode === 'into' && allowedChildNodeTypes.length)
```

Comment:
Just looking at the code, this condition could be made more efficient.
`(preferredMode === 'before' || preferredMode === 'after' || preferredMode === 'into') && allowedChildNodeTypes.length)

Reply:
Huh? But that's a wrong logic?

Reply:
Upps, sorry, ignore me, of course `allowedSiblingNodeTypes` and `allowedChildNodeTypes` are different things! sorry!

Reply:
Heh, it was late at night...